hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -414,71 +540,130 @@ Since the life cycle partition is the only partition that needs live updates in-
 The life cycle state is hence encoded such that incremental updates to the state are always carried out at the granularity of a 16bit word.
 Further, the life cycle transition counter is encoded such that each stroke consumes a full 16bit word for the same reason.
 
-See [**TODO: add link to lifecycle spec and state encoding**] for more details on the life cycle encoding.
+See [life cycle controller documentation]({{< relref "hw/ip/lc_ctrl/doc" >}}) for more details on the life cycle encoding.
 
-### Partition Implementations
+### Partition Controllers
 
-There are three different partition types that are reused throughout the implementation:
+In RTL, we distinguish between buffered and unbuffered partition modules.
+These are parameterized, such that we can assemble the array of OTP partitions with these two modules only.
+The corresponding controller FSMs are explained in more detail below.
 
-1. Unbuffered partition
-2. Buffered partition
-3. Life cycle partition
+#### Unbuffered Partition
 
-Each partition type has a different RTL implementation, but what they have in common is that each partition has its own controller and set of local buffer registers (if required).
+![Unbuffered Partition FSM](otp_ctrl_unbuf_part_fsm.svg)
 
-#### Unbuffered Partition
+As shown above, the unbuffered partition module has a relatively simple controller FSM that only reads out the digest value of the partition upon initialization, and then basically waits for TL-UL read transactions to its corresponding window in the CSR space.
 
-**TODO: add FSM details**
+Write access through the DAI will be locked in case the digest is set to a non-zero value.
+Also, read access through the DAI and the CSR window can be locked at runtime via a CSR.
+Read transactions through the CSR window will error out if they are out of bounds, or if read access is locked.
+
+Note that unrecoverable [OTP errors]({{< relref "#generalized-open-source-interface" >}}) or parity failures in the digest register will move the partition controller into a terminal error state.
 
 #### Buffered Partition
 
-**TODO: add FSM details**
+![Buffered Partition FSM](otp_ctrl_buf_part_fsm.svg)
+
+The controller FSM of the buffered partition module is more complex than the unbuffered counterpart, since it has to account for scrambling and digest calculation.
+
+Upon initialization, the controller reads out the whole partition and descrambles it on the fly if needed.
+
+Then, right after the initial readout, the partition controller jumps into the first integrity check, which behaves somewhat differently, depending on whether the partition is digest protected and scrambled.
```

Comment:
"depending on whether the partition is digest protected (or not) and/or scrambled (or not)"

That's still awkward but to use "whether" in this way you need an alternative case, so at least one "or not" - you really have 3-4 cases here so you could also do:

"depending on whether the partition is digest protected and scrambled, digest protected and unscrambled, without digest protection and scrambled, or without digestion protection and unscrambled"

(this is also kinda clunky to me but choose whatever you prefer)

Reply:
thanks, I've used your first suggestion, since it is a bit more concise.